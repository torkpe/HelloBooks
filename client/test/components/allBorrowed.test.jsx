import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import connectedAllBorrowed, { AllBorrowed } from '../../components/AllBorrowed.jsx';
import storeConfiguration from '../../store/store';

configure({ adapter: new Adapter() });

describe('<AllBorrowed />', () => {
  const props = {
    getAllBorrowed: jest.fn(),
    clearBooksState: jest.fn(),
    auth: {
      user: {
        id: 1
      },
    },
    getAllBorrowedBooks: {
      borrowedBooks: {
        books: [
          {
            title: 'eze goes to school',
            description: 'a boy from the village who finally went to school',
            id: 1,
            cover: 'xcgvhjkhcvbkjhgcvbjhcv',
            pdf: 'fghjkhgfghjgfgg',
            quantity: 4,
            genre: 'ghjgv'
          },
          {
            title: 'eze goes to school',
            description: 'a boy from the village who finally went to school',
            id: 1,
            cover: 'xcgvhjkhcvbkjhgcvbjhcv',
            pdf: 'fghjkhgfghjgfgg',
            quantity: 4,
            genre: 'ghjgv'
          }
        ]
      }
    },
    loading: false,
  };
  const mockStore = configureMockStore();
  let shallowComponent;
  let store;
  let wrapper;
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    shallowComponent = shallow(<AllBorrowed {...props} />);
  });
  it('renders <AllBorrowed /> component', () => {
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
  it('calls clearBooksState', () => {
    shallowComponent = shallow(<AllBorrowed {...props} />);
    shallowComponent.instance().componentWillUnmount();
    expect(shallowComponent.instance().props.clearBooksState).toHaveBeenCalled();
  });
  it('checks if loading is true', () => {
    props.loading = true;
    shallowComponent = shallow(<AllBorrowed {...props} />);
    expect(shallowComponent.instance().props.loading).toEqual(true);
  });
});
