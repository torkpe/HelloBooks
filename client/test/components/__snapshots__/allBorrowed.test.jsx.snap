// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllBorrowed /> renders <AllBorrowed /> component 1`] = `
<div
  className="mdl-grid"
>
  <div
    className="mdl-cell mdl-cell--1-col"
  />
  <div
    className="mdl-cell mdl-cell--10-col"
  >
    <div
      className="mdl-grid"
    >
      <div
        className="mdl-cell mdl-cell--2-col"
      />
    </div>
    <div
      className="ask"
    >
      <h5>
         All Borrowed Books 
      </h5>
      <hr />
    </div>
    <div
      className="ask"
    />
    <div
      className="mdl-grid"
    >
      <div
        className="contents"
      >
        <h5>
          You have no record of borrowed books
        </h5>
      </div>
    </div>
    <div
      className="mdl-grid "
    >
      <div
        className="mdl-cell mdl-cell--2-col"
      />
    </div>
  </div>
  <div
    className="mdl-cell mdl-cell--1-col"
  />
</div>
`;
